src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs

```diff
@@ -486,20 +486,16 @@ private bool RemoveTransaction(Transaction transaction)
             // Check the inputs - include those that have a reference to a transaction containing one of our scripts and the same index.
             foreach (TxIn input in transaction.Inputs)
             {
-                if (!this.outpointLookup.TryGetValue(input.PrevOut, out TransactionData tTx))
+                if (!this.outpointLookup.TryGetValue(input.PrevOut, out TransactionData spentTransaction))
                 {
                     continue;
                 }
 
                 // Get the transaction being spent and unspend it.
-                TransactionData spentTransaction = this.Wallet.MultiSigAddress.Transactions.SingleOrDefault(t => (t.Id == tTx.Id) && (t.Index == tTx.Index));
-                if (spentTransaction != null)
-                {
-                    this.logger.LogTrace("Unspending {0}-{1}", spentTransaction.Id, spentTransaction.Index);
+                this.logger.LogTrace("Unspending {0}-{1}", spentTransaction.Id, spentTransaction.Index);
```

Comment:
OK but above we get it from the outpointLookup and below from the MultiSigAddress transaction collection?

Reply:
I.e. in the previous code we first check `outpointLookup` and then we check `this.Wallet.MultiSigAddress.Transactions` ?

Reply:
See this:

`private Dictionary<OutPoint, TransactionData> outpointLookup => this.Wallet.MultiSigAddress.Transactions.GetOutpointLookup();`

Reply:
Ah ok yes I see now :)